tests/test-admin-gpg.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+#
+# Copyright (C) 2019 Rafael Fonseca <r4f4rfs@gmail.com>
+#
+# SPDX-License-Identifier: LGPL-2.0+
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+set -euo pipefail
+
+. $(dirname $0)/libtest.sh
+
+setup_os_repository_signed () {
+    mode=$1
+    shift
+    bootmode=$1
+    shift
+    bootdir=${1:-usr/lib/modules/3.6.0}
+
+    oldpwd=`pwd`
+    keyid="472CDAFA"
+
+    cd ${test_tmpdir}
+    mkdir testos-repo
+    if test -n "$mode"; then
+	      ostree_repo_init testos-repo --mode=${mode}
+    else
+	      ostree_repo_init testos-repo
+    fi
+
+    cd ${test_tmpdir}
+    mkdir osdata
+    cd osdata
+    kver=3.6.0
+    mkdir -p usr/bin ${bootdir} usr/lib/modules/${kver} usr/share usr/etc
```

Comment:
Future enhancement: Let's avoid duplicating the code for setup (maybe make gpg signatures a flag)?

Reply:
:+1: Agree - my thought was for this test it'd be okay as `setup_os_repository_signed` is local to this file, but to enhance in future, the one in `libtest.sh` could be updated with a flag.